pkg/workloadmeta/collectors/internal/containerd/container_builder.go

```diff
@@ -100,3 +111,40 @@ func extractStatus(status containerd.ProcessStatus) workloadmeta.ContainerStatus
 
 	return workloadmeta.ContainerStatusUnknown
 }
+
+// extractIP gets the IP of a container.
+//
+// The containerd client does not expose the IPs, that's why we use the helpers
+// that we have in the "system" package to extract that information from
+// "/proc".
+//
+// A current limitation is that if a container exposes multiple IPs, this
+// function just returns one of them. That means that if a container is attached
+// to multiple networks this might not work as expected.
+func extractIP(container containerd.Container, containerdClient cutil.ContainerdItf) (string, error) {
+	taskPids, err := containerdClient.TaskPids(container)
+	if err != nil {
+		return "", err
+	}
+
+	if len(taskPids) == 0 {
+		return "", errors.New("no PIDs found")
+	}
+
+	IPs, err := system.ParseProcessIPs(
```

Comment:
This only works on Linux, thus I believe that this function should go to a `_linux.go` file, which will remove the need for `pkg/util/system/network_stub.go`.

Also this is not guaranteed to work, it requires that host `/proc` is mounted, see https://github.com/DataDog/datadog-agent/blob/main/pkg/util/containers/v2/metrics/system/collector_linux.go#L51-L59

Reply:
I moved the file in this commit: https://github.com/DataDog/datadog-agent/pull/11924/commits/889003658a350cdf24b3091831733f7d947920f8

I've also modified the function so that it checks `/proc` as the first step. This way the limitation that it needs to be mounted is more clear: https://github.com/DataDog/datadog-agent/pull/11924/commits/10a7c829fd95d3bf75e2a110fb3b63202a2f27ae
If `/proc` is not mounted, an error is logged (with debug level), but the container is still stored in the workloadmeta store.